### ISSUE TITLE
Strange behavior when `scrollbar-size` is set to `0 0`

### ISSUE THREAD

**Issue by kraanzu** [Textualize/textual#3886]
> Have you checked closed issues? https://github.com/Textualize/textual/issues?q=is%3Aissue+is%3Aclosed

Yes

> Please give a brief but clear explanation of the issue. If you can, include a complete working example that demonstrates the bug. **Check it can run without modifications.**

So when the `scrollbar-size` is set to `0 0`, the content of the widget just vanishes

```python
from textual.app import App, ComposeResult
from textual.widgets import Static

class TestApp(App):
    DEFAULT_CSS = """
    Screen {
        scrollbar-size: 0 0;
    }
    """

    def compose(self) -> ComposeResult:
        yield Static("Hello, world!\n" * 100)

TestApp().run()
```

However, if set to `0 1` the code works just fine

It will be helpful if you run the following command and paste the results:

```
textual diagnose
```

# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.45.1 |
| Rich    | 13.7.0 |

## Python

| Name           | Value                                             |
|----------------|---------------------------------------------------|
| Version        | 3.11.6                                            |
| Implementation | CPython                                           |
| Compiler       | GCC 13.2.1 20230801                               |
| Executable     | /home/kraanzu/Documents/termtyper/venv/bin/python |

## Operating System

| Name    | Value                                                  |
|---------|--------------------------------------------------------|
| System  | Linux                                                  |
| Release | 6.6.2-arch1-1                                          |
| Version | #1 SMP PREEMPT_DYNAMIC Mon, 20 Nov 2023 23:18:21 +0000 |

## Terminal

| Name                 | Value                              |
|----------------------|------------------------------------|
| Terminal Application | WezTerm (20230712-072601-f4abf8fd) |
| TERM                 | xterm-256color                     |
| COLORTERM            | truecolor                          |
| FORCE_COLOR          | *Not set*                          |
| NO_COLOR             | *Not set*                          |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=144, height=34 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 144                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 34                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

Feel free to add screenshots and / or videos. These can be very helpful!

**Comment by willmcgugan** [comment]
Thanks. Fixes in 0.46.0

**Comment by kraanzu** [comment]
Awesome! Thanks for the quick fix :D
